### ISSUE TITLE
LinearGaussianBayesianNetwork class should be merged with BayesianModel

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#751]
The whole idea to keep the network structure and parameters separate from each other was to deal with these special cases. Since `LinearGaussianBayesianNetwork` is just a special case with `LinearGaussianCPD`,  so using `BayesianModel` we should be able to define the network structure and add `LinearGaussianCPD` to it.

**Comment by yashu-seth** [comment]
@ankurankan Yes, but currently BayesianModel only supports discrete variables. I think we will have different classes for discrete and continuous variables for the Bayesian Models.

**Comment by ankurankan** [comment]
@yashu-seth I don't think that we will need separate classes. But I haven't thought about the exact implementation details yet. Any reasons why you think we will need separate classes ?

**Comment by yashu-seth** [comment]
@ankurankan I have also not figured out the exact details. Lets just keep it like this for now. It would be more feasible to first decide how we want to represent the continuous and discrete cases before we decide what needs to be done about `LinearGaussianBayesianNetwork`.

**Comment by ankurankan** [comment]
@yashu-seth My idea of treating continuous and discrete case is to use the same model structure for both. The only difference would be for parameterization of the models and for that we have two separate directories in factors. And for inference we should handle both the cases internally, so that user has the same API/methods for both the cases.

**Comment by yashu-seth** [comment]
@ankurankan That seems a feasible solution. Let me also think this through, I will be in a better position to comment then.

**Comment by yashu-seth** [comment]
@ankurankan I think having a common class for handling continuous, discrete and hybrid models would be the best option. It will also give a common API to the users. Do you plan on working on this? Or should I look into this? 
I will merge ```LinearGaussianBayesianNetwork``` into ```BayesianModel```. and will also figure out a way to keep a common API for inference and model fitting.

**Comment by ankurankan** [comment]
@yashu-seth Yes, that's the best way (not sure how difficult would it be add hybrid models). Sure go ahead. Also since you are at it, you might also look into this: #763

**Comment by linzhp** [comment]
If we are going to use the same class for both discrete and continuous CPDs, how to deal with [get_cardinality](https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/models/BayesianModel.py#L318) method?

**Comment by ankurankan** [comment]
@linzhp We can just have a check over the kind of CPDs in the network and throw an exception if has continuous CPDs.
